### PR TITLE
Electron-77 (User config)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
Implemented method to update user config on installer.

@VishwasShashidhar , @VikasShashidhar & @lneir Please review. Thanks.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -118,6 +131,37 @@ function setStartup(lStartup){
     });
 }
 
+// Method to overwrite user config on mac installer
```
Comment (VishwasShashidhar) [inline js/main.js]: Any reason we are creating a child process here in place of using node's fs api for manipulating files?

File: js/main.js [hunk 2]
```diff
@@ -118,6 +131,37 @@ function setStartup(lStartup){
...
+function updateUserConfigMac() {
```
Comment (VishwasShashidhar) [inline js/main.js]: Any reason we are creating a child process here in place of using node's fs api for manipulating files?

File: js/main.js [hunk 3]
```diff
@@ -118,6 +131,37 @@ function setStartup(lStartup){
...
+            if (err) {
+                reject(err);
+            }
+            resolve();
+        });
+    });
+}
+
+// Method to overwrite user config on windows installer
+function updateUserConfigWin() {
```
Comment (VishwasShashidhar) [inline js/main.js]: We should use a common method for mac and windows. Just is the isMac flag to check and take action accordingly.

File: js/main.js (hunk 2 above)
Comment (KiranNiranjan) [inline js/main.js]: I tried implementing this with the `node's fs` module and the following is the issue that I faced.
 
The installer runs as a root process where the directory created will be with a different permissions so to change the permissions using the method `fs.chown(path, uid, gid, callback)` requires `uid - integer` & `gid - integer` but we don't get these values if the process is running in root.

**`Root User`**
screen shot 2017-07-14 at 8 57 33 pm

**`User`**
screen shot 2017-07-14 at 8 58 23 pm

**Comment by VikasShashidhar** [comment]
@KiranNiranjan @VishwasShashidhar Is this ready?

**Comment by KiranNiranjan** [comment]
This feature is working fine with the `child_process`. But there are issues using the `fs` module.

**Comment by VishwasShashidhar** [comment]
In that case, we can go ahead with the child process implementation. lgtm.

**Pull request merged**